File: app/src/main/java/io/eodc/planit/db/DateConverter.java
package io.eodc.planit.db;

import android.arch.persistence.room.TypeConverter;

import org.joda.time.DateTime;

class DateConverter {
    @TypeConverter
    public static DateTime fromTimestamp(Long time) {
        return time == null ? null : new DateTime(time);
    }

    @TypeConverter
    public static Long toTimestamp(DateTime time) {
        return time == null ? null : time.getMillis();
    }
}

File: app/src/main/java/io/eodc/planit/db/DateConverter.kt
package io.eodc.planit.db

import android.arch.persistence.room.TypeConverter

import org.joda.time.DateTime

class DateConverter {
    @TypeConverter
    fun fromTimestamp(time: Long?): DateTime? {
        return if (time == null) null else DateTime(time)
    }

    @TypeConverter
    fun toTimestamp(time: DateTime?): Long? {
        return time?.millis
    }
}
